tests/test_invenio_accounts.py

```diff
@@ -136,6 +136,27 @@ def test_datastore_rolecreate(app):
         assert 1 == Role.query.filter_by(name="superuser").count()
 
 
+def test_datastore_update_role(app):
+    """Test update role."""
+    ds = app.extensions["invenio-accounts"].datastore
+
+    with app.app_context():
+        r1 = ds.create_role(id="1", name="superuser", description="1234")
+        ds.commit()
+        r2 = ds.find_role("superuser")
+        assert r1 == r2
+        assert 1 == Role.query.filter_by(name="superuser").count()
+
+        r1 = ds.update_role(
+            Role(id="1", name="megauser", description="updated description")
+        )
+        ds.commit()
+        r2 = ds.find_role("megauser")
+        assert r1 == r2
+        assert r2.description == "updated description"
+        assert 1 == Role.query.filter_by(name="megauser").count()
```

Comment:
I would check that superuser does not exist anymore, to verify is an update not an addition.

```python
assert 0 == Role.query.filter_by(name="superuser").count()
```